java-frontend/src/main/java/org/sonar/java/model/JUtils.java

```diff
@@ -300,4 +305,34 @@ public static boolean hasUnknownTypeInHierarchy(Symbol.TypeSymbol typeSymbol) {
     return hasUnknownTypeInHierarchy(superClass.symbol());
   }
 
+  /**
+   * @param typeSymbol to lookup recursively for unknown symbol on itself, its interfaces and its super classes hierarchy.
+   * @param methodSymbol a method having at least one unknown parameter type that prevent the overriding resolution
+   *                    to find any exact match. This methodSymbol will be used to find a potential match, just by
+   *                    comparing the name and parameter count.
+   * @return true if the given typeSymbol and its inheritance hierarchy is 100% known and no method in the inheritance
+   * cloud match (name and parameter count) the given methodSymbol. In this case, we are sure the methodSymbol does not
+   * override any another methods.
+   */
+  private static boolean hasUnknownTypeInHierarchyOrAnyMatchingMethod(Symbol.TypeSymbol typeSymbol, Symbol.MethodSymbol methodSymbol) {
+    if (typeSymbol.isUnknown()) {
+      return true;
+    }
+    if (typeSymbol != methodSymbol.owner() && typeSymbol.memberSymbols().stream()
+      .anyMatch(member -> methodSymbol.name().equals(member.name()) &&
```

Comment:
Very minor: I believe it would improve efficiency if we did the String comparison last

Reply:
I did the change, but not in a "Fix for review" separated commit, sorry for the reviewer. I wanted to keep my 3 commits as clean as possible and ready to be merged.